Kernel and rtc

georgemoralis opened the pull request:
1. Added sceRtcGetCurrentTick

georgemoralis committed:
added dummy rtc module

georgemoralis committed:
added sceRtcGetCurrentTick

georgemoralis committed:
stubbed event_flag functions

georgemoralis committed:
event flags : added function signatures

georgemoralis committed:
sceKernelCreateEventFlag , sceKernelWaitEventFlag implementation

georgemoralis committed:
eventflags enums

raziel1000 committed:
- scePthreadCondTimedwait
- gettimeofday
- sceRtcGetCurrentClockLocalTime
- sceRtcGetDayOfWeek

georgemoralis committed:
added some posix thread calls

psucien commented on src/core/libraries/kernel/event_flag/event_flag.cpp:
```diff
@@ -0,0 +1,141 @@
+    bool fifo = true;
+
+    switch (attr) {
+    case 0x10:
```
ORBIS_KERNEL_EVF_ATTR_SINGLE ?

georgemoralis replied:
? it is createEventFlag not wait..

georgemoralis replied:
yes i will make it more clear :D

raziel1000 committed:
scePthreadCondDestroy, posix_pthread_mutexattr_settype, pthread_self(posix)

raziel1000 committed:
-scePthreadRwlockRdlock
-scePthreadRwlockWrlock
-scePthreadRwlockUnlock
-scePthreadRwlockDestroy
-scePthreadRwlockattrInit
-posix_pthread_rwlock_rdlock
-posix_pthread_rwlock_unlock

raziel1000 committed:
posix_pthread_mutexattr_setprotocol
posix_pthread_mutex_trylock
posix_pthread_condattr_init
posix_pthread_condattr_destroy
ps4_gettimeofday (posix)

georgemoralis committed:
improved scePthreadCondTimedwait

georgemoralis committed:
Merge branch 'kernel_and_rtc' of https://github.com/shadps4-emu/shadPS4 into kernel_and_rtc

raziel1000 committed:
Added:
sceKernelGetCpumode
posix_getpagesize
posix_usleep
sceKernelSleep
ps4_clock_getres
posix_pthread_setspecific
posix_pthread_key_create
scePthreadKeyCreate
scePthreadRename
scePthreadJoin

Improved:
sceKernelUsleep
sceKernelNanosleep

raziel1000 committed:
Added:
- sceSaveDataMount
- Create2 mode (50)
- slight change in error handling (testing for now)
- log to print message when fd = 1 in sceKernelWrite (to avoid crash and move forward, not a file)
- mount savedata at emu start for CUSA00402 - Crimsonland (only case, needs testing)
- sceSaveDataGetMountInfo

- scePthreadGetthreadid
- posix_getpid
- posix_pthread_attr_setdetachstate
- posix_pthread_attr_destroy
- posix_pthread_detach
- scePthreadSetschedparam(partial)
- posix_pthread_setschedparam (partial)
- posix_sched_get_priority_max
